wp-admin/js/widgets/media-widgets.js

```diff
@@ -713,6 +713,20 @@ wp.mediaWidgets = ( function( $ ) {
 				modelProps.attachment_id = mediaFrameProps.id;
 			}
 
+			isEmbed = 0 === modelProps.attachment_id;
+
+			// Reset all video/audio sources.
+			wp.media.view.settings.embedExts.map( function( ext ) {
+
+				// Only reset when the main source changes.
+				if ( ! ( ext in modelProps ) || isEmbed ) {
+					modelProps[ ext ] = '';
+				}
+			} );
+
+			extension = mediaFrameProps.url.split( '.' ).pop();
+			modelProps[ extension ] = mediaFrameProps.url;
```

Comment:
Shouldn't this only be done if `isEmbed`? Also, if `! isEmbed` then I believe that `modelProps.url = ''` needs to be done, so that we reserve `url` for external embeds and then the extension props for media uploaded to the media library?